app/controllers/posts_controller.rb

```diff
@@ -54,28 +56,19 @@ def redirect_from_purchase_id
   def send_for_purchase
     authorize @post
 
-    purchase = current_seller.sales.find_by_external_id!(params[:purchase_id])
-
-    # Limit the number of emails sent per post to avoid abuse.
-    Rails.cache.fetch("post_email:#{@post.id}:#{purchase.id}", expires_in: 8.hours) do
-      CreatorContactingCustomersEmailInfo.where(purchase:, installment: @post).destroy_all
-
-      PostEmailApi.process(
-        post: @post,
-        recipients: [
-          {
-            email: purchase.email,
-            purchase:,
-            url_redirect: purchase.url_redirect,
-            subscription: purchase.subscription,
-          }.compact_blank
-        ])
-      true
-    end
+    SendPostsForPurchaseService.send_post(post: @post, purchase: @purchase)
```

Comment:
Unrelated to resend all but part of re-factor. This endpoint resends email for when "Resend" is clicked

<img width="669" height="780" alt="Image" src="https://github.com/user-attachments/assets/bbee9bfa-878b-44d4-83b4-1d50c9c9a0c7" />